docs/source/checkpoints/dist_ckpt.rst

```diff
@@ -0,0 +1,392 @@
+
+ShardedTensorFactory
+--------------------
+This class allows to defer tensors transformations until the actual saving.
+A factory can expand a tensor into an arbitrary sub state dict (including all supported entities listed above).
+The need for such deferral will be explained in the `Tensors transformations`_ section.
+
+LocalNonpersitentObject
+-----------------------
+This is a simple wrapper that allows to express the fact that the object wrapped with this class should end up in the final loaded state dict during loading.
+During saving such objects are ignored.
+
+Arbitrary object
+----------------
+All objects different than dicts, lists and the instances of the classes listed above are treated as "common" objects.
+
+During saving, all such objects in the sharded state dict passed to `dist_checkpointing.save` are assumed to be duplicated across ranks and therefore saved only by a single coordinator rank (rank 0).
+
+During loading, all such objects in the sharded state dict passed to `dist_checkpointing.load` are simply ignored - the loaded state dict contains only "common" objects that are were actually saved in the checkpoint.
+
+
+
+
+Entrypoints
+===========
+There are several useful user entrypoints for checkpoint saving and loading.
+
+dist_checkpointing.save
+-----------------------
+The only entrypoint for checkpoint saving.
+Requires providing a sharded state dict to save and saving strategies for handling different entities (see `Save and load strategies`_ for detailed explanation).
+The sharded state dict is processed in the following way:
+
+1. The ShardedTensorFactories are applied
+2. LocalNonPersistentObject are extracted from the sharded state dict and ignored
+3. ShardedBase objects are extracted
+4. All other objects are treated as "common" and saved according to a sharded strategy (see `Save and load strategies`_)
+5. All ShardedObjects are extracted from point (3) objects and saved with a common strategy (see `Save and load strategies`_)
+6. All ShardedTensors are saved.
+7. `metadata.json` file with backend and version metadata is saved to the checkpoint directory.
+
+dist_checkpointing.load
+-----------------------
+The main entrypoint for checkpoint loading.
+Requires providing a sharded state dict (in order to implicitly define mappings between local tensors and checkpoint tensors) and loading strategies.
+In practice, the same sharded state dict can be usually used for both saving and loading (the sharded state dict for loading will just contain tensors with uninitialized data).
+
+The sharded state dict provided as an input is processed in the following way:
+
+1. "common" state dict is loaded from the checkpoint. This forms the base of the resulting state dict
+2. The ShardedTensorFactories from the input sharded state dict are applied
+3. LocalNonPersistentObject are extracted from the input sharded state dict, unwrapped and added to the resulting state dict
```

Comment:
3. The LocalNonPersistentObject is extracted from the input sharded state dict, unwrapped and added to the resulting state dict.